File: An 2/Sem 1/MAP/Lab7/Java/src/lab7/Model/Student.java
package lab7.Model;

import java.io.Serializable;

/**
 *
 * @author mihai
 */
public class Student implements Comparable<Student>, Readable<Student>, Serializable {
    public String name;
    public int id;
    public int grade;

    public Student() {

    }

    public Student(int id, String name, int grade){
        this.name = name;
        this.id = id;
        this.grade = grade;
    }

    public float average() {
        return grade;
    }

    @Override
    public boolean isGreaterThan(Student student) {
        return (this.grade > student.grade);
    }

    @Override
    public String toString() {
        return String.format("%s|%d|%s|%d|", this.getClass().toString(), this.id, this.name, this.grade);
    }

    @Override
    public void readAttributesFromString(String string) {
        String[] tokens = string.split("[|]+");
        this.id = Integer.parseInt(tokens[1]);
        this.name = tokens[2];
        this.grade = Integer.parseInt(tokens[3]);
    }
}

File: An 2/Sem 1/MAP/Lab10/C#/Lab2/Model/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab2
{
	namespace Lab2_Model {
		[Serializable()]
		public class Student : Comparable<Student>, Readable, HasId
	    {
            public string name { get; set; }
            public int id { get; set; }
            public int grade { get; set; }


			public Student() {
			}

	        public Student(int id, string name, int grade)
	        {
	            this.id = id;
	            this.name = name;
	            this.grade = grade;
	        }

			public virtual float average() {
				return grade;
			}

			bool Comparable<Student>.isGreaterThan(Student student) {
				return (this.grade > student.grade);
			}

			public override String ToString() {
				return String.Format("{0}|{1}|{2}|{3}|", this.GetType(), this.id, this.name, this.grade);
			}

			void Readable.readAttributesFromString(String inputString) {
				string[] tokens = inputString.Split ('|');
				this.id = Convert.ToInt16 (tokens [1]);
				this.name = tokens [2];
				this.grade = Convert.ToInt16 (tokens [3]);
			}

			int HasId.getId() {
				return this.id;
			}
	    }
	}
}
